samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/GradingService.java

```diff
@@ -912,6 +909,23 @@ public int compare(ItemGradingData o1, ItemGradingData o2) {
 	      });
       }
       
+    //IMAGEMAP_QUESTION - order by itemGradingId if it is an imageMap question
+      if (isImageMapQuestion(tempItemGradinglist, publishedItemHash)) {
+	      Collections.sort(tempItemGradinglist, new Comparator<ItemGradingData>(){
+	    	  public int compare(ItemGradingData o1, ItemGradingData o2) {
```

Comment:
It would be ideal to externalize this comparator so that in can be unit tested.
Also see springs NullSafeComparator or use java's newer lambda comparator which deals with nulls in a better way either is fine.
Example org.sakaiproject.util.comparator.UserSortNameComparator or https://www.baeldung.com/java-8-comparator-comparing

Reply:
I created a class, but we need compare Longs, so I just moved it to a file. How do you see it?